### PR TITLE
SOC-5406 Optimize database queries for activity stream

### PR DESCRIPTION
**Pull request by boubaker**
First optimization of JPQL used to get activities

### CODE REVIEW

File: lib/src/main/java/org/exoplatform/social/addons/storage/entity/ActivityEntity.java [hunk 1]
```diff
@@ -57,41 +57,33 @@
                 query = "SELECT a FROM SocActivity a WHERE a.hidden = false  AND a.updatedDate > :sinceTime AND exists (SELECT item FROM a.streamItems item WHERE item.ownerId in (:owner)) ORDER BY a.updatedDate ASC"),
         @NamedQuery(name = "SocActivity.getActivityByOwnerAndProviderId",
                 query = "SELECT a FROM SocActivity a WHERE a.hidden = false AND a.providerId = :providerId AND exists (SELECT item FROM a.streamItems item WHERE item.ownerId in (:owner)) ORDER BY a.updatedDate DESC"),
+        @NamedQuery(name = "SocActivity.getConnections",
+                query = "SELECT conn1.receiver.id from SocConnection conn1 where conn1.status = :connStatus  and conn1.receiver.id = :identityId" +
+                        " union " +
+                        "SELECT conn2.sender.id from SocConnection conn2 where conn2.status = :connStatus and conn2.sender.id = :identityId "),
```
Comment (nttuyen) [inline lib/src/main/java/org/exoplatform/social/addons/storage/entity/ActivityEntity.java]: We should put this query on ConnectionEntity class, I think. It's easier to maintain and follow. We should rename it to "SocConnection.getConnections" also.

If I remember well, JPA 2.0 (the version we are using) does not support union in query. We could use this kind of query with Hibernate implementation?

**Comment by nttuyen** [comment]
I still don't know why this query could be much more quickly without add more configuration on MySQL Server. 

As Neji reported, this query still very slowly on this comment: https://jira.exoplatform.org/browse/SOC-5406?focusedCommentId=407820&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-407820

File: lib/src/main/java/org/exoplatform/social/addons/storage/entity/ActivityEntity.java (hunk 1 above)
Comment (boubaker) [inline lib/src/main/java/org/exoplatform/social/addons/storage/entity/ActivityEntity.java]: Looks better to move to ConnectionEntity, yes.
I don't have a problem to use "union" using Hibernate (I don't think we will want to transparently migrate to another JPA impl). If this is not convincing you, I can switch to use Standard SQL, WDYT ?

Comment (thomasdelhomenie) [inline lib/src/main/java/org/exoplatform/social/addons/storage/entity/ActivityEntity.java]: Using Hibernate specific feature is fine if it is not supported by JPA. The recommendation is of course to use JPA as much as we can, but we can use Hibernate specific feature if necessary.

**Comment by thomasdelhomenie** [comment]
I don't either. Same dataset ?
It has been validated by the last TQA tests, so let's consider it as validated.
Boubaker is doing some final tests to make sure everything is ok then we will merge it.

**Comment by boubaker** [comment]
Merged

**Pull request closed without merging**